src/metadata.rs

```diff
@@ -254,6 +254,15 @@ impl Drop for Metadata {
     }
 }
 
+impl From<*const grpc_metadata_array> for Metadata {
```

Comment:
Instead of implementing `From` trait, you can implement a new c function to take the metadata. There is already a function `grpcwrap_metadata_array_move`, so `grpcwrap_batch_context_recv_initial_metadata` can be implemented as
```
GPR_EXPORT void GPR_CALLTYPE
grpcwrap_batch_context_take_initial_metadata(
    const grpcwrap_batch_context* ctx, grpc_metadata_array* initial_metadata) {
    grpcwrap_metadata_array_move(initial_metadata, &(ctx->send_initial_metadata));
}
```

And in rust side, just declare a new metadata and pass it to c:
```
let mut metadata = MetadataBuilder::new().build();
unsafe {
    grpcwrap_batch_context_take_initial_metadata(ctx, &mut metadata);
}
```

. So that we don't expose unsafe implementation to safe type.